internal/pkg/dependencies/common.go

```diff
@@ -79,10 +79,18 @@ type clientWithToken struct {
 	Token *storageapi.Token
 }
 
-func (v *CommonContainer) WithStorageApiClient(client client.Client, token *storageapi.Token) *CommonContainer {
+func (v *CommonContainer) WithStorageApiClient(c client.Client, token *storageapi.Token) *CommonContainer {
 	clone := *v
+
+	// Reset all values dependent on Storage API Project/Token
 	clone.storageApi = new(Lazy[clientWithToken])
-	clone.storageApi.Set(clientWithToken{Client: client, Token: token})
+	clone.storageApiIndex = new(Lazy[storageapi.Index])
+	clone.features = new(Lazy[storageapi.FeaturesMap])
+	clone.schedulerApi = new(Lazy[client.Client])
+	clone.eventSender = new(Lazy[event.Sender])
```

Comment:
Tu som nasiel bug, ... ked chceme nastavit storage api, tak musime vycistit vsetky hodnoty, ktore na nom zavisia.
Najprv je tam storage api, do ktoreho nie sme prihlaseny, ... daju sa nacitat `components`, `services` ... ale ako nahla sa prihlasime, tak napr. `index` endpoint vrati uz `features` pre konkretny projekt.

Reply:
(ked sa vratim z dovo, tak tie dependencies chcem trocha lepsie premysliet, aby to bol prehladnejsie)